colour/models/rgb/transfer_functions/gamma.py

```diff
@@ -51,8 +68,27 @@ def gamma_function(a, exponent=1.0):
     --------
     >>> gamma_function(0.18, 2.2)  # doctest: +ELLIPSIS
     0.0229932...
+    >>> gamma_function(-0.18, 2.0)  # doctest: +ELLIPSIS
+    0.0323999...
+    >>> gamma_function(-0.18, 2.2)
+    nan
+    >>> gamma_function(-0.18, 2.2, 'Mirror')  # doctest: +ELLIPSIS
+    -0.0229932...
+    >>> gamma_function(-0.18, 2.2, 'Preserve')  # doctest: +ELLIPSIS
+    -0.1...
+    >>> gamma_function(-0.18, 2.2, 'Clamp')  # doctest: +ELLIPSIS
+    0.0
     """
 
     a = np.asarray(a)
+    exponent = np.asarray(exponent)
 
-    return a ** exponent
+    negative_number_handling = negative_number_handling.lower()
+    if negative_number_handling == 'indeterminate':
+        return as_numeric(a ** exponent)
+    elif negative_number_handling == 'mirror':
+        return as_numeric(np.sign(a) * np.abs(a) ** exponent)
+    elif negative_number_handling == 'preserve':
+        return as_numeric(np.where(a < 0, a, a ** exponent))
+    elif negative_number_handling == 'clamp':
+        return as_numeric(np.where(a < 0, 0, a ** exponent))
```

Comment:
Should we throw an exception in the final else that indicates that the `negative_number_handling` had an invalid value?